### ISSUE TITLE
NPU环境训练速度问题

### ISSUE THREAD

**Issue by obj12** [modelscope/ms-swift#3331]
在NPU 910B3上共4卡环境，使用GRPO算法训练qwen2.5-1.5b，数据集来源参照的swift给的GRPO示例流程(zouxuhong/Countdown-Tasks-3to4)。训练速度不知道是否为正常速度？

训练脚本：
source /usr/local/Ascend/ascend-toolkit/set_env.sh
source /usr/local/Ascend/nnal/atb/set_env.sh

# export PYTORCH_NPU_ALLOC_CONF=expandable_segments:True
export PYTORCH_NPU_ALLOC_CONF=expandable_segments:False
export HCCL_RDMA_TIMEOUT=20 # 修复错误

export MASTER_ADDR=localhost
export VLLM_HOST_IP=localhost

ASCEND_RT_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=2 \
swift rlhf \
    --rlhf_type grpo \
    --model /home/ma-user/work/hjx_deploy/qwen/qwen_2.5/1.5b \
    --model_type 'qwen2' \
    --external_plugins examples/train/grpo/plugin/plugin.py \
    --reward_funcs external_countdown format \
    --use_vllm true \
    --vllm_enable_prefix_caching false \
    --vllm_device auto \
    --vllm_gpu_memory_utilization 0.9 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset '/home/ma-user/work/hjx_rlhf/swift/ms-swift/data#10000' \
    --max_length 2048 \
    --max_completion_length 1024 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 2 \
    --learning_rate 5e-7 \
    --gradient_accumulation_steps 2 \
    --eval_steps 500 \
    --save_steps 100 \
    --save_total_limit 20 \
    --logging_steps 1 \
    --output_dir output/GRPO_COUNTDOWN \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 4 \
    --num_generations 4 \
    --temperature 1.0 \
    --system 'You are a helpful assistant. You first thinks about the reasoning process in the mind and then provides the user with the answer.' \
    --deepspeed zero3 \
    --log_completions true \
    --vllm_max_model_len 1024 \
    --beta 0.001 \
    --num_iterations 1 > grpo_1.5_log.txt 2>&1 &

训练日志：
{"loss": 0.10042733, "grad_norm": 3.03476667, "learning_rate": 1e-08, "train_speed(iter/s)": 0.008267, "completion_length": 318.125, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.625, "reward": 0.625, "reward_std": 0.53867513, "kl": 0.0, "clip_ratio": 0.0, "epoch": 0.00020202, "global_step/max_steps": "1/4950", "percentage": "0.02%", "elapsed_time": "1m 13s", "remaining_time": "4d 4h 52m 0s"}
{"loss": -0.10017388, "grad_norm": 4.18892717, "learning_rate": 2e-08, "train_speed(iter/s)": 0.011187, "completion_length": 269.375, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.75, "reward": 0.75, "reward_std": 0.5, "kl": 0.0, "clip_ratio": 0.0, "epoch": 0.00040404, "global_step/max_steps": "2/4950", "percentage": "0.04%", "elapsed_time": "2m 11s", "remaining_time": "3d 18h 9m 4s"}
{"loss": 0.14865024, "grad_norm": 2.10723329, "learning_rate": 3e-08, "train_speed(iter/s)": 0.012452, "completion_length": 250.875, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.75, "reward": 0.75, "reward_std": 0.28867513, "kl": 0.00043392, "clip_ratio": 0.0, "epoch": 0.00060606, "global_step/max_steps": "3/4950", "percentage": "0.06%", "elapsed_time": "3m 13s", "remaining_time": "3d 16h 32m 58s"}
{"loss": 0.24209723, "grad_norm": 2.26306248, "learning_rate": 4e-08, "train_speed(iter/s)": 0.0131, "completion_length": 315.875, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.125, "rewards/Format": 0.75, "reward": 0.875, "reward_std": 0.65824831, "kl": 0.00028563, "clip_ratio": 0.0, "epoch": 0.00080808, "global_step/max_steps": "4/4950", "percentage": "0.08%", "elapsed_time": "4m 17s", "remaining_time": "3d 16h 31m 32s"}
{"loss": -0.01671183, "grad_norm": 6.54592896, "learning_rate": 5e-08, "train_speed(iter/s)": 0.014826, "completion_length": 131.125, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.75, "reward": 0.75, "reward_std": 0.5, "kl": 0.00041771, "clip_ratio": 0.0, "epoch": 0.0010101, "global_step/max_steps": "5/4950", "percentage": "0.10%", "elapsed_time": "4m 49s", "remaining_time": "3d 7h 34m 12s"}
{"loss": 0.29984978, "grad_norm": 2.40538764, "learning_rate": 6e-08, "train_speed(iter/s)": 0.014155, "completion_length": 335.125, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.125, "rewards/Format": 0.875, "reward": 1.0, "reward_std": 0.5, "kl": 0.00034428, "clip_ratio": 0.0, "epoch": 0.00121212, "global_step/max_steps": "6/4950", "percentage": "0.12%", "elapsed_time": "6m 16s", "remaining_time": "3d 14h 7m 38s"}
{"loss": 0.01343828, "grad_norm": 3.40092087, "learning_rate": 7e-08, "train_speed(iter/s)": 0.014477, "completion_length": 301.375, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.625, "reward": 0.625, "reward_std": 0.53867513, "kl": 0.00018597, "clip_ratio": 0.0, "epoch": 0.00141414, "global_step/max_steps": "7/4950", "percentage": "0.14%", "elapsed_time": "7m 15s", "remaining_time": "3d 13h 30m 27s"}
{"loss": 2.4e-07, "grad_norm": 0.00089076, "learning_rate": 8e-08, "train_speed(iter/s)": 0.015525, "completion_length": 204.5, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.125, "rewards/Format": 0.875, "reward": 1.0, "reward_std": 0.0, "kl": 0.00024414, "clip_ratio": 0.0, "epoch": 0.00161616, "global_step/max_steps": "8/4950", "percentage": "0.16%", "elapsed_time": "7m 47s", "remaining_time": "3d 8h 15m 13s"}
{"loss": 0.13877492, "grad_norm": 3.66878939, "learning_rate": 9e-08, "train_speed(iter/s)": 0.014859, "completion_length": 392.625, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.375, "reward": 0.375, "reward_std": 0.53867513, "kl": 0.00023365, "clip_ratio": 0.0, "epoch": 0.00181818, "global_step/max_steps": "9/4950", "percentage": "0.18%", "elapsed_time": "9m 18s", "remaining_time": "3d 13h 6m 34s"}
{"loss": 0.2252617, "grad_norm": 2.67573428, "learning_rate": 1e-07, "train_speed(iter/s)": 0.015003, "completion_length": 316.875, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.375, "reward": 0.375, "reward_std": 0.53867513, "kl": 0.00019932, "clip_ratio": 0.0, "epoch": 0.0020202, "global_step/max_steps": "10/4950", "percentage": "0.20%", "elapsed_time": "10m 18s", "remaining_time": "3d 12h 56m 2s"}
{"loss": 0.24143943, "grad_norm": 3.22009158, "learning_rate": 1.1e-07, "train_speed(iter/s)": 0.014872, "completion_length": 315.75, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.25, "rewards/Format": 0.5, "reward": 0.75, "reward_std": 0.69692343, "kl": 0.00027204, "clip_ratio": 0.0, "epoch": 0.00222222, "global_step/max_steps": "11/4950", "percentage": "0.22%", "elapsed_time": "11m 32s", "remaining_time": "3d 14h 18m 55s"}
{"loss": 2.7e-07, "grad_norm": 2.80426526, "learning_rate": 1.2e-07, "train_speed(iter/s)": 0.015381, "completion_length": 193.75, "response_clip_ratio": 0.0, "rewards/CountdownORM": 0.0, "rewards/Format": 0.75, "reward": 0.75, "reward_std": 0.28867513, "kl": 0.00035763, "clip_ratio": 0.0, "epoch": 0.00242424, "global_step/max_steps": "12/4950", "percentage": "0.24%", "elapsed_time": "12m 12s", "remaining_time": "3d 11h 44m 19s"}
 
系统信息&软件版本
ms-swift: main branch
device: Ascend 910B3
CANN vesion: 8.1.RC
ATB version:  8.1.RC1
pip list:
torch                  2.5.1
torch-npu              2.5.1.dev20250218
transformers           4.49.0
trl                    0.16.0.dev0
vllm                   0.7.1+empty
vllm_ascend            0.7.1rc1
deepspeed              0.16.4
